contracts/tokenomics/gauges/sidechain/RootGaugeEvmos.vy

```diff
@@ -0,0 +1,253 @@
+# @version 0.3.2
+"""
+@title Root-Chain Gauge
+@author Curve Finance
+@license MIT
+@notice Calculates total allocated weekly SDL emission
+        mints and sends across a sidechain bridge
+"""
+
+from vyper.interfaces import ERC20
+
+interface Controller:
+    def period() -> int128: view
+    def gauge_relative_weight(addr: address, time: uint256) -> uint256: view
+    def checkpoint(): nonpayable
+    def checkpoint_gauge(addr: address): nonpayable
+
+interface Minter:
+    def token() -> address: view
+    def controller() -> address: view
+    def minted(user: address, gauge: address) -> uint256: view
+    def mint(gauge: address): nonpayable
+    def start_epoch_time_write() -> uint256: nonpayable
+    def rate() -> uint256: view
+    def committed_rate() -> uint256: view
+
+
+event PeriodEmission:
+    period_start: uint256
+    mint_amount: uint256
+
+event CommitOwnership:
+    admin: address
+
+event ApplyOwnership:
+    admin: address
+
+event FeesModified:
+    gas_limit: uint256
+    gas_price: uint256
+    max_submission_cost: uint256
+
+
+## BRIDGE AND L2 SPECIFIC CONSTANTS 
+
+BRIDGE_ROUTER: constant(address) = 0x88A69B4E698A4B090DF6CF5Bd7B2D47325Ad30A3
+NOMAD_EVMOS_MAINNET_DESTINATION_CODE: constant(uint32) = 1702260083
+NOMAD_BRIDGE_ENABLE_FAST_TRANSFER: constant(bool) = False 
+NAME: immutable(String[64])
```

Comment:
Along with the immutable constant `NAME`, we also need to add a consistent name function/variable with ERC20 ABI.

In this case that would be a function with lowercase like below.

```vyper
@view
@external
def name() -> String[64]:
  """
  @notice Get the name for this gauge token
  """
  return NAME
```

Reply:
Would we? The rootgauges don't implement the ERC20 interface. Or does frontend need a dedicated getter function?